### PR TITLE
Add RAM/ROM stats to the application

### PR DESCRIPTION
**Pull request by deepikabhavnani**
Merge after 5.11 is released

### CODE REVIEW

File: main.cpp [hunk 1]
```diff
@@ -41,5 +41,16 @@ int main()
     */
     printf("Compiler Version: %d \n", stats.compiler_version);
 
+    /* RAM / ROM memory start and size information */
+    for (int i = 0; i < MBED_MAX_MEM_REGIONS; i++) {
+        if (stats.ram_size[i] != 0) {
+            printf("RAM%d: Start 0x%x Size: 0x%x \n",i , stats.ram_start[i], stats.ram_size[i]);
```
Comment (kegilbert) [inline main.cpp]: Got a warning from casting a uint32_t for ram_start/size to %x in printf, changed to long x and tweaked a bit of whitespace.
```suggestion
            printf("RAM%d: Start 0x%lx Size: 0x%lx \n", i, stats.ram_start[i], stats.ram_size[i]);
```

File: main.cpp [hunk 2]
```diff
@@ -41,5 +41,16 @@ int main()
...
+        }
+    }
+    for (int i = 0; i < MBED_MAX_MEM_REGIONS; i++) {
+        if (stats.rom_size[i] != 0) {
+            printf("ROM%d: Start 0x%x Size: 0x%x \n",i, stats.rom_start[i], stats.rom_size[i]);
```
Comment (kegilbert) [inline main.cpp]: Got a warning from casting a uint32_t for ram_start/size to %x in printf, changed to long x and tweaked a bit of whitespace.
```suggestion
            printf("ROM%d: Start 0x%lx Size: 0x%lx \n", i, stats.rom_start[i], stats.rom_size[i]);
```

**Comment by deepikabhavnani** [comment]
@kegilbert - Please review. 
Updated mbed-os lib to master which includes new changes to API, good to merge

**Review by kegilbert — APPROVED** [review]
LGTM!

**Pull request merged**